doc/source/operations/baremetal-node-management.rst

```diff
@@ -0,0 +1,277 @@
+
+.. code-block:: bash
+
+    openstack baremetal port list --field uuid --field node_uuid --field local_link_connection --format value
+
+NGS will manage VLAN membership for ports when the ``local_link_connection`` fields match one of the switches in ``ml2_conf.ini``.
+The rest of the switch configuration is static.
+The switch configuration that NGS will apply to these ports is detailed in :ref:`dynamic-switch-configuration`.
+
+.. _ironic-node-discovery:
+
+Ironic node discovery
+---------------------
+
+Discovery is a process used to automatically enrol new nodes in Ironic.
+It works by PXE booting the nodes into the Ironic Python Agent (IPA) ramdisk.
+This ramdisk will collect hardware and networking configuration from the node in a process known as introspection.
+This data is used to populate the baremetal node object in Ironic.
+The series of steps you need to take to enrol a new node is as follows:
+
+- Configure credentials on the BMC. These are needed for Ironic to be able to perform power control actions.
+
+- Controllers should have network connectivity with the target BMC.
+
+- (If kayobe manages physical network) Add any additional switch configuration to kayobe config.
+  The minimal switch configuration that kayobe needs to know about is described in :ref:`tor-switch-configuration`.
+
+- Apply any :ref:`static switch configration <static-switch-config>`. This performs the initial
+  setup of the switchports that is needed before Ironic can take over. The static configuration
+  will not be modified by Ironic, so it should be safe to reapply at any point. See :ref:`ngs-commands`
+  for details about the switch configuation that Networking Generic Switch will apply.
+
+- (If kayobe manages physical network) Put the node onto the provisioning network by using the
+  ``--enable-discovery`` flag and either ``--interface-description-limit`` or ``--interface-limit``
+  (do not run this command without one of these limits). See :ref:`static-switch-config`.
+
+    * This is only necessary to initially discover the node. Once the node is in registered in Ironic,
+      it will take over control of the the VLAN membership. See :ref:`dynamic-switch-configuration`.
+
+    * This provides ethernet connectivity with the controllers over the `workload provisioning` network
+
+- (If kayobe doesn't manage physical network) Put the node onto the provisioning network.
+
+.. TODO: link to the relevant file in kayobe config
+
+- Add node to the kayobe inventory.
+
+.. TODO: Fill in details about necessary BIOS & RAID config
+
+- Apply any necesary BIOS & RAID configuration.
+
+.. TODO: Fill in details about how to trigger a PXE boot
+
+- PXE boot the node.
+
+- If the discovery process is successful, the node will appear in Ironic and will get populated with the necessary information from the hardware inspection process.
+
+.. TODO: Link to the Kayobe inventory in the repo
+
```

Comment:
Question: Can those details in TODOs be generalised as well? Or should we keep this in admin guide?

Reply:
Either would work. General instructions could describe the most common setups e.g. how to trigger a PXE boot on the most common hardware, when describing the kayobe inventory, it could give a few example paths for where those file might be. It won't be as precise as custom per-deployment instructions but that is quite high-effort